acceptance/tests/i18n/modules/puppet_agent_with_module.rb

```diff
@@ -0,0 +1,145 @@
+  confine :except, :platform => /^aix/ # QENG-5283 needed for this to work
+
+  tag 'audit:medium',
+      'audit:acceptance'
+
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  require 'puppet/acceptance/i18n_utils'
+  extend Puppet::Acceptance::I18nUtils
+
+  require 'puppet/acceptance/i18ndemo_utils'
+  extend Puppet::Acceptance::I18nDemoUtils
+
+  language = 'ja_JP'
+  step "configure server locale to #{language}" do
+    configure_master_system_locale(language)
+  end
+
+  app_type          = File.basename(__FILE__, '.*')
+  tmp_environment_1 = mk_tmp_environment_with_teardown(master, app_type)
+
+  step 'install a i18ndemo module' do
+    install_i18n_demo_module(master, tmp_environment_1)
+  end
+
+  teardown do
+    step 'resetting the server locale' do
+      reset_master_system_locale
+    end
+  end
+
+  agents.each do |agent|
+
+    step("ensure #{language} locale is configured") do
+      language = enable_locale_language(agent, language)
+      skip_test("test machine is missing #{language} locale. Skipping") if language.nil?
+    end
+
+    type_path = agent.tmpdir('provider')
+    teardown do
+      on(agent, "rm -rf '#{type_path}'")
+    end
+
+    step "Run puppet agent of a module with language #{language} and verify the translations" do
+      step 'verify custom fact translations' do
+        site_pp_content_1 = <<-PP
+          node default {
+            include i18ndemo
+          }
+        PP
+        create_sitepp(master, tmp_environment_1, site_pp_content_1)
+        on(agent, puppet("agent -t --server #{master.hostname} --environment #{tmp_environment_1}"), 'ENV' => { 'LANGUAGE' => language }, :acceptable_exit_codes => [0, 2]) do |result|
+          assert_match(/Error:.*eputnam-i18ndemo fact: これはeputnam-i18ndemoからのカスタムファクトからのレイズです/, result.stderr, 'missing translation for raise from ruby fact')
+        end
+      end
+
+      step 'verify translations from init.pp' do
+        site_pp_content_2 = <<-PP
+          node default {
+            class {'i18ndemo': }
+          }
+        PP
+        create_sitepp(master, tmp_environment_1, site_pp_content_2)
+        on(agent, puppet("agent -t --server #{master.hostname} --environment #{tmp_environment_1}"), 'ENV' => { 'LANGUAGE' => language }) do |result|
+          assert_match(/Warning:.*eputnam-i18ndemo init.pp: ファイルの作成/, result.stderr, 'missing warning translation from init.pp')
+        end
+
+        site_pp_content_3 = <<-PP
+          node default {
+            class { 'i18ndemo': param1 => false }
+          }
+        PP
+        create_sitepp(master, tmp_environment_1, site_pp_content_3)
+        # TODO
+        # TODO
+        # TODO
+        # TODO are we supposed to get a Server 500 on SERVER: error here?
+        # Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Function Call,
```

Comment:
Not sure, but it should be translated (both "Server Error" and the "Failed to create").

Reply:
Yes, this failure is on purpose. They should both be translated, the "failed to create" message comes from the module in init.pp.